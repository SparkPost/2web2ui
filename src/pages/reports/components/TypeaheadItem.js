import React from 'react';

import styles from './Typeahead.module.scss';

const TypeaheadItem = ({ value, helpText = null }) => (
  <div className={styles.Item}>
    <span className={styles.Value}>{ value }</span>
    { helpText && <span className={styles.HelpText}>{ helpText }</span> }
  </div>
);

export default TypeaheadItem;
