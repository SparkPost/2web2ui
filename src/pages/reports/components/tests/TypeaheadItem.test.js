import React from 'react';
import { shallow } from 'enzyme';
import TypeaheadItem from '../TypeaheadItem';

describe('Component: TypeaheadItem', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      value: 'found'
    };
    wrapper = shallow(<TypeaheadItem {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render help text', () => {
    wrapper.setProps({ helpText: 'I am helping!' });
    expect(wrapper).toMatchSnapshot();
  });
});
