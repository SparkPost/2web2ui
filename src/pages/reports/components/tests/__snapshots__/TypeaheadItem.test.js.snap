// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TypeaheadItem should render 1`] = `
<div
  className="Item"
>
  <span
    className="Value"
  >
    found
  </span>
</div>
`;

exports[`Component: TypeaheadItem should render help text 1`] = `
<div
  className="Item"
>
  <span
    className="Value"
  >
    found
  </span>
  <span
    className="HelpText"
  >
    I am helping!
  </span>
</div>
`;
